File: OPENAPI-REST-API/swagger-client/dart/lib/model/inline_response20013_filter_firewall_filters.dart
part of swagger.api;

class InlineResponse20013FilterFirewallFilters {
  
  String daddr = null;

  String dest = null;

  String filterName = null;

  String destinationIp = null;

  String filter = null;

  InlineResponse20013FilterFirewallFilters();

  @override
  String toString() {
    return 'InlineResponse20013FilterFirewallFilters[daddr=$daddr, dest=$dest, filterName=$filterName, destinationIp=$destinationIp, filter=$filter, ]';
  }

  InlineResponse20013FilterFirewallFilters.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    daddr = json['daddr'];
    dest = json['dest'];
    filterName = json['filter_name'];
    destinationIp = json['destination_ip'];
    filter = json['filter'];
  }

  Map<String, dynamic> toJson() {
    return {
      'daddr': daddr,
      'dest': dest,
      'filter_name': filterName,
      'destination_ip': destinationIp,
      'filter': filter
     };
  }

  static List<InlineResponse20013FilterFirewallFilters> listFromJson(List<dynamic> json) {
    return json == null ? new List<InlineResponse20013FilterFirewallFilters>() : json.map((value) => new InlineResponse20013FilterFirewallFilters.fromJson(value)).toList();
  }

  static Map<String, InlineResponse20013FilterFirewallFilters> mapFromJson(Map<String, Map<String, dynamic>> json) {
    var map = new Map<String, InlineResponse20013FilterFirewallFilters>();
    if (json != null && json.length > 0) {
      json.forEach((String key, Map<String, dynamic> value) => map[key] = new InlineResponse20013FilterFirewallFilters.fromJson(value));
    }
    return map;
  }
}

File: OPENAPI-REST-API/swagger-client/micronaut/src/main/java/io/swagger/model/InlineResponse20013FilterFirewallFilters.java
package io.swagger.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import io.swagger.v3.oas.annotations.media.Schema;
import io.micronaut.core.annotation.Introspected;
import io.micronaut.validation.Validated;
import javax.validation.Valid;
import javax.validation.constraints.*;

/**
 * InlineResponse20013FilterFirewallFilters
 */
@Validated
@Introspected

public class InlineResponse20013FilterFirewallFilters   {
  @JsonProperty("daddr")
  private String daddr = null;

  @JsonProperty("dest")
  private String dest = null;

  @JsonProperty("filter_name")
  private String filterName = null;

  @JsonProperty("destination_ip")
  private String destinationIp = null;

  @JsonProperty("filter")
  private String filter = null;

  public InlineResponse20013FilterFirewallFilters daddr(String daddr) {
    this.daddr = daddr;
    return this;
  }

  /**
   * Get daddr
   * @return daddr
  **/
  @Schema(description = "")
  @NotNull

  public String getDaddr() {
    return daddr;
  }

  public void setDaddr(String daddr) {
    this.daddr = daddr;
  }

  public InlineResponse20013FilterFirewallFilters dest(String dest) {
    this.dest = dest;
    return this;
  }

  /**
   * Get dest
   * @return dest
  **/
  @Schema(description = "")
  @NotNull

  public String getDest() {
    return dest;
  }

  public void setDest(String dest) {
    this.dest = dest;
  }

  public InlineResponse20013FilterFirewallFilters filterName(String filterName) {
    this.filterName = filterName;
    return this;
  }

  /**
   * Get filterName
   * @return filterName
  **/
  @Schema(description = "")
  @NotNull

  public String getFilterName() {
    return filterName;
  }

  public void setFilterName(String filterName) {
    this.filterName = filterName;
  }

  public InlineResponse20013FilterFirewallFilters destinationIp(String destinationIp) {
    this.destinationIp = destinationIp;
    return this;
  }

  /**
   * Get destinationIp
   * @return destinationIp
  **/
  @Schema(description = "")
  @NotNull

  public String getDestinationIp() {
    return destinationIp;
  }

  public void setDestinationIp(String destinationIp) {
    this.destinationIp = destinationIp;
  }

  public InlineResponse20013FilterFirewallFilters filter(String filter) {
    this.filter = filter;
    return this;
  }

  /**
   * Get filter
   * @return filter
  **/
  @Schema(description = "")
  @NotNull

  public String getFilter() {
    return filter;
  }

  public void setFilter(String filter) {
    this.filter = filter;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    InlineResponse20013FilterFirewallFilters inlineResponse20013FilterFirewallFilters = (InlineResponse20013FilterFirewallFilters) o;
    return Objects.equals(this.daddr, inlineResponse20013FilterFirewallFilters.daddr) &&
        Objects.equals(this.dest, inlineResponse20013FilterFirewallFilters.dest) &&
        Objects.equals(this.filterName, inlineResponse20013FilterFirewallFilters.filterName) &&
        Objects.equals(this.destinationIp, inlineResponse20013FilterFirewallFilters.destinationIp) &&
        Objects.equals(this.filter, inlineResponse20013FilterFirewallFilters.filter);
  }

  @Override
  public int hashCode() {
    return Objects.hash(daddr, dest, filterName, destinationIp, filter);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class InlineResponse20013FilterFirewallFilters {\n");
    
    sb.append("    daddr: ").append(toIndentedString(daddr)).append("\n");
    sb.append("    dest: ").append(toIndentedString(dest)).append("\n");
    sb.append("    filterName: ").append(toIndentedString(filterName)).append("\n");
    sb.append("    destinationIp: ").append(toIndentedString(destinationIp)).append("\n");
    sb.append("    filter: ").append(toIndentedString(filter)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}
